packages/tasks/src/tasks/mask-generation/about.md

```diff
@@ -6,25 +6,53 @@ When filtering for an image, the generated masks might serve as an initial filte
 
-### Human-in-the-loop
+### Human-in-the-loop Computer Vision Applications
 
-For applications where humans are in the loop, masks highlight certain region of images for humans to validate.
+For applications where humans are in the loop, masks highlight certain regions of images for humans to validate.
 
 ## Task Variants
 
 ### Segmentation
 
-Image Segmentation divides an image into segments where each pixel in the image is mapped to an object. This task has multiple variants such as instance segmentation, panoptic segmentation and semantic segmentation. You can learn more about segmentation on its [task page](https://huggingface.co/tasks/image-segmentation).
+Image Segmentation divides an image into segments where each pixel is mapped to an object. This task has multiple variants, such as instance segmentation, panoptic segmentation, and semantic segmentation. You can learn more about segmentation on its [task page](https://huggingface.co/tasks/image-segmentation).
 
 ## Inference
 
+Mask generation models often work in two modes: segment everything or prompt mode. 
+Below, you can see segment everything mode where many masks will be returned. 
+
 ```python
 from transformers import pipeline
-generator =  pipeline("mask-generation", device = 0, points_per_batch = 256)
+
+generator =  pipeline("mask-generation", model = "Zigeng/SlimSAM-uniform-50", points_per_batch = 64, device = "cuda")
 image_url = "https://huggingface.co/ybelkada/segment-anything/resolve/main/assets/car.png"
-outputs = generator(image_url, points_per_batch = 256)
+outputs = generator(image_url, points_per_batch = 64)
+outputs["masks"]
+# array of multiple binary masks returned for each generated mask
+```
+
+The prompt mode takes in three types of prompts.
+- **Point prompt:** The user can leave a point on the image, and a meaningful segment around the point will be returned.
+- **Box prompt:** The user can draw a box on the image, and a meaningful segment within the box will be returned.
+- **Text prompt:** The user can input a text, and the objects of that type will be segmented. Note that this capability has not yet been released and has only been explored in research.
+
+Below you can see how to input point prompting with model inference without `pipeline`. 
+```python
+from transformers import SamModel, SamProcessor
+from PIL import Image
+import requests 
+
+model = SamModel.from_pretrained("Zigeng/SlimSAM-uniform-50").to("cuda")
+processor = SamProcessor.from_pretrained("Zigeng/SlimSAM-uniform-50")
+
+raw_image = Image.open(requests.get(image_url, stream=True).raw).convert("RGB")
+input_points = [[[450, 600]]]
+inputs = processor(raw_image, input_points=input_points, return_tensors="pt").to("cuda")
+outputs = model(**inputs)
+masks = processor.image_processor.post_process_masks(outputs.pred_masks.cpu(), inputs["original_sizes"].cpu(), inputs["reshaped_input_sizes"].cpu())
```

Comment:
Is there a direct path from the processor to `post_process_masks`?

Reply:
sorry this is my fault I think you're right